cmd/diagramdemo/main.go

```diff
@@ -0,0 +1,137 @@
+package main
+
+import (
+	"fmt"
+	"image/color"
+	"time"
+
+	"fyne.io/x/fyne/widget/diagramwidget"
+
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/app"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/widget"
+)
+
+var forceticks int = 0
+
+func forceanim() {
+
+	// XXX: very naughty -- accesses shared memory in potentially unsafe
```

Comment:
IIRC I did this because I wanted a way to see the layout changing in realtime for debugging purposes. That said, this could probably be made safe if we wrapped forceticks in a mutex.

Reply:
Charles: can you be more specific about your thoughts for the Mutex - what is being locked?